CMakeLists.txt

```diff
@@ -1,100 +1,81 @@
 cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 project(chip_8_cpp VERSION 1.0.0)
 
-if(NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE Release)
-endif()
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/modules/")
+file(COPY assets DESTINATION ${CMAKE_BINARY_DIR})
 
-add_definitions(-DASSETS_PATH="${PROJECT_SOURCE_DIR}/assets/")
+set(MAIN_TARGET ${PROJECT_NAME}_interpreter)
+add_executable(${MAIN_TARGET} src/main.cpp ${INTERPRETER_SOURCE_FILES})
 
-find_package(SFML 2.3 COMPONENTS graphics audio window system)
+set_property(TARGET ${MAIN_TARGET} PROPERTY CXX_STANDARD 17)
+set_property(TARGET ${MAIN_TARGET} PROPERTY CXX_STANDARD_REQUIRED TRUE)
 
-list(APPEND INTERPRETER_SOURCE_FILES
-    src/Interpreter.cpp
-    src/io/Speaker.cpp
-    src/io/Keypad.cpp
-    src/io/KeyMap.cpp
-    src/io/display/PixelArray.cpp
-    src/io/display/Renderer.cpp
-    src/details/memory.cpp
-    src/details/audio.cpp
-)
+target_include_directories(${MAIN_TARGET} PRIVATE "${PROJECT_SOURCE_DIR}/include")
 
-set(EXECUTABLE_NAME ${PROJECT_NAME}_interpreter)
+include(compiler-warnings)
+target_add_compiler_warnigs(${MAIN_TARGET})
 
-if(OS_ARCH)
-    add_executable(${EXECUTABLE_NAME} src/main.cpp ${INTERPRETER_SOURCE_FILES})
-    target_include_directories(${EXECUTABLE_NAME} PRIVATE include)
-    target_compile_options(${EXECUTABLE_NAME} PRIVATE --std=c++17 -Wall -Werror -Wfatal-errors -pedantic)
-    find_package(SFML 2.5 COMPONENTS graphics audio window system REQUIRED)
-    target_link_libraries(${EXECUTABLE_NAME} sfml-graphics sfml-audio sfml-window sfml-system)
-endif()
+include(sfml)
+target_add_sfml(${MAIN_TARGET})
 
-if(OS_UBUNTU)
-    set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/modules/")
-    find_package(SFML 2.3 COMPONENTS graphics audio window system)
-    add_executable(${PROJECT_NAME}_interpreter src/main.cpp ${INTERPRETER_SOURCE_FILES})
-    target_include_directories(${PROJECT_NAME}_interpreter PRIVATE include ${SFML_INCLUDE_DIRS})
-    target_link_libraries(${PROJECT_NAME}_interpreter PRIVATE ${SFML_LIBRARIES})
-    target_compile_options(${PROJECT_NAME}_interpreter PRIVATE --std=c++17 -Wall -Werror -Wfatal-errors -pedantic)
-endif()
+include(clang-tidy)
+target_add_clang_tidy(${MAIN_TARGET})
+
+include(include-what-you-use)
+set(CMAKE_LINK_WHAT_YOU_USE TRUE)
+
+target_sources(${MAIN_TARGET}
+    PRIVATE
```

Comment:
Sources are listed again in the test section below. Why don't you just use the `list` command? Is it to avoid using a variable?

Reply:
No variables :P. Check the updated version, allows setting properties very nicely and now it doesn't need to repeat.